pkg/installer/custominstallconfig.go

```diff
@@ -147,12 +162,117 @@ func (m *manager) applyInstallConfigCustomisations(installConfig *installconfig.
+	// Adding etc host files.
+	etcHostIgnConfig, err := EtcHostsIgnitionConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP)
+	if err != nil {
+		return nil, err
+	}
+	etcHostMasterMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "master")
+	if err != nil {
+		return nil, err
+	}
+	data, err = yaml.Marshal(etcHostMasterMachineConfig)
+	if err != nil {
+		return nil, err
+	}
+	etcMasterConfig := ignition.FileFromBytes("/opt/openshift/openshift/99_openshift-machineconfig_99-master-aro-etc-hosts-gateway-domains.yaml", "root", 0644, data)
+	etcHostWorkerMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "worker")
+	if err != nil {
+		return nil, err
+	}
+	data, err = yaml.Marshal(etcHostWorkerMachineConfig)
+	if err != nil {
+		return nil, err
+	}
+	etcWorkerConfig := ignition.FileFromBytes("/opt/openshift/openshift/99_openshift-machineconfig_99-worker-aro-etc-hosts-gateway-domains.yaml", "root", 0644, data)
 
 	bootstrapAsset := g.Get(&bootstrap.Bootstrap{}).(*bootstrap.Bootstrap)
 	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, dnsmasqIgnConfig.Storage.Files...)
 	bootstrapAsset.Config.Systemd.Units = append(bootstrapAsset.Config.Systemd.Units, dnsmasqIgnConfig.Systemd.Units...)
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, masterConfig)
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, workerConfig)
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, etcHostIgnConfig.Storage.Files...)
+	bootstrapAsset.Config.Systemd.Units = append(bootstrapAsset.Config.Systemd.Units, etcHostIgnConfig.Systemd.Units...)
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, etcMasterConfig)
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, etcWorkerConfig)
+	err = appendMdsdFiles(bootstrapAsset, bootstrapLoggingConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	baseManifestPath := "opt/openshift/manifests"
+	config := struct {
+		AROWorkerRegistries string
+		HTTPSecret          string
+		AccountName         string
+		ContainerName       string
+		CloudName           string
+		AROIngressInternal  bool
+		AROIngressIP        string
+	}{
+		AROWorkerRegistries: aroWorkerRegistries(installConfig.Config.DeprecatedImageContentSources),
+		HTTPSecret:          imageRegistryConfig.HTTPSecret,
+		AccountName:         imageRegistryConfig.AccountName,
+		ContainerName:       imageRegistryConfig.ContainerName,
+		CloudName:           imageRegistryConfig.CloudName,
+		AROIngressInternal:  false,
+		AROIngressIP:        dnsConfig.IngressIP,
+	}
+	err = filepath.Walk(baseManifestPath,
```

Comment:
This appears to be reading from the local filesystem, so it only works when running from the source directory.